src/main/java/com/mongodb/hibernate/dialect/MongoDialectSettings.java

```diff
@@ -0,0 +1,293 @@
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+import org.hibernate.cfg.AvailableSettings;
+import org.hibernate.service.spi.Configurable;
+import org.jspecify.annotations.Nullable;
+
+/**
+ * The configuration of {@link MongoDialect}, {@link MongoConnectionProvider}, the internal {@link MongoClient}.
+ *
+ * <table>
+ *     <caption>Supported configuration properties</caption>
+ *     <thead>
+ *         <tr>
+ *             <th>Method</th>
+ *             <th>Has default</th>
+ *             <th>Related {@linkplain Configurable#configure(Map) configuration property} name</th>
+ *             <th>Supported value types of the configuration property</th>
+ *             <th>Value, unless overridden via {@link Builder}</th>
+ *         </tr>
+ *     </thead>
+ *     <tbody>
+ *         <tr>
+ *             <td>{@link #getMongoClientSettings()}</td>
+ *             <td>✓</td>
+ *             <td>{@value AvailableSettings#JAKARTA_JDBC_URL}</td>
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}</li>
+ *                     <li>{@link ConnectionString}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>Is {@linkplain MongoClientSettings.Builder#applyConnectionString(ConnectionString) based} on
+ *             the {@link ConnectionString} {@linkplain ConnectionString#ConnectionString(String) constructed} from
+ *             {@value AvailableSettings#JAKARTA_JDBC_URL}, if the latter is configured; otherwise a {@link MongoClientSettings}
+ *             instance with its defaults.</td>
+ *         </tr>
+ *         <tr>
+ *             <td>{@link #getDatabaseName()}</td>
+ *             <td>✗</td>
+ *             <td>{@value AvailableSettings#JAKARTA_JDBC_URL}</td>
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}</li>
+ *                     <li>{@link ConnectionString}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>The MongoDB database name from {@value AvailableSettings#JAKARTA_JDBC_URL},
+ *             if {@linkplain ConnectionString#getDatabase() configured};
+ *             otherwise there is no default value, and a value must be configured via {@link Builder#databaseName(String)}.</td>
+ *         </tr>
+ *         <tr>
+ *             <td>{@link #isAutoCommit()}</td>
+ *             <td>✓</td>
+ *             <td>{@value AvailableSettings#AUTOCOMMIT}</td>
+ *             <td>
+ *                 <ul>
+ *                     <li>{@link String}, only {@code "true"}, {@code "false"} are allowed</li>
+ *                     <li>{@link Boolean}</li>
+ *                 </ul>
+ *             </td>
+ *             <td>{@value AvailableSettings#AUTOCOMMIT}, if configured; otherwise {@code false}.</td>
+ *         </tr>
+ *     </tbody>
+ * </table>
+ *
+ * @see MongoDialectConfigurator
+ */
+public final class MongoDialectSettings {
```

Comment:
This API is named and designed after `MongoClientSettings`. Given that it allows an application to access `MongoClientSettings.Builder`, this approach seems reasonable to me.

Reply:
With the design as is, we don't need to expose neither `MongoDialectSettings` nor `MongoDialectSettings.builder` as part of our API. However, hiding them requires taking an approach different from that of the `MongoClientSettings`. I think, I am fine with the approach I propose in this PR (because I am trying to make `MongoDialectSettings` similar to `MongoClientSettings` here). But if there are concerns about exposing `MongoDialectSettings`, please bring them up.